packages/dds/map/src/interfaces.ts

```diff
@@ -48,6 +48,12 @@ export interface IDirectory extends Map<string, any>, IEventProvider<IDirectoryE
      */
     set<T = any>(key: string, value: T): this;
 
+    /**
+     * Get the number of sub directory within the directory.
+     * @returns The number of sub directory within a directory.
+     */
+    countSubDirectory?(): number;
```

Comment:
I do not like optionality - it makes it hard to use it right away.
I think (for our DDSs) we should assume that our DDSs are the only concrete implementations of these interfaces, and thus addition is Ok.
It's not different from adding new API.

Reply:
FFX repo does implement the IDirectory here: 
https://office.visualstudio.com/OC/_git/office-bohemia?path=/packages/conversa/src/store/primitives/ObservableDirectory.ts&version=GBmaster&line=22&lineEnd=22&lineStartColumn=1&lineEndColumn=7&lineStyle=plain&_a=contents
So, I am not sure what should be the right way here.

Reply:
my only worry is making this non-optional in main could break partners tests if they do any mocking to validate their business logic. ideally users never see build breaks due to patch updates.

Reply:
i think optional is the right way in main, and you can immediately make it non-optional in next, and add a breaking note

Reply:
I know it's not a full solution of the problem, but in C++ world we usually generated mocks as part of implementation. They were generic / "empty", where anybody can overwrite any method (usually not through inheritance, but through setMockMetod_X() kind of APIs).
Not providing mocks results in multiple very similar implementations, even in our code base.

Reply:
it looks like the existing implementation isn't even a mock. more of an extension